proposals/NNNN-generalize-async-sequence.md

```diff
@@ -0,0 +1,311 @@
+# Generalize `AsyncSequence` and `AsyncIteratorProtocol`
+
+* Proposal: [SE-NNNN](NNNN-generalize-async-sequence.md)
+* Authors: [Doug Gregor](https://github.com/douggregor), [Holly Borla](https://github.com/hborla)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Implementation: https://github.com/apple/swift/pull/70635
+* Review: ([pitch](https://forums.swift.org/t/pitch-generalize-asyncsequence-and-asynciteratorprotocol/69283))
+
+## Introduction
+
+This proposal generalizes `AsyncSequence` in two ways:
+1. The bespoke `@rethrows` attribute is replaced with proper `throws` polymorphism via adoption of typed throws.
```

Comment:
I do wonder how much discussion of `@rethrows` is appropriate given that it's never been an official feature. Perhaps makes sense to mention as an aside below but I don't know if it warrants callout up-front?

Reply:
I changed this to just say proper `throws` polymorphism is accomplished with typed throws adoption.